service/clients/oracle/client.go

```diff
@@ -93,23 +92,51 @@ func NewClient(
 		return nil, fmt.Errorf("timeout must be positive")
 	}
 
-	return &GRPCClient{
+	client := &GRPCClient{
 		logger:  logger,
 		addr:    addr,
 		timeout: timeout,
 		metrics: metrics,
-	}, nil
+	}
+
+	// apply options
+	for _, opt := range opts {
+		opt(client)
+	}
+
+	return client, nil
 }
 
 // Start starts the GRPC client. This method dials the remote oracle-service
-// and errors if the connection fails.
-func (c *GRPCClient) Start() error {
+// and errors if the connection fails. This method may block (depending on the blockingDial option).
+func (c *GRPCClient) Start(ctx context.Context) error {
 	c.logger.Info("starting oracle client", "addr", c.addr)
 
-	conn, err := grpc.Dial(
-		c.addr,
+	opts := []grpc.DialOption{
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
+
+	if c.blockingDial {
+		opts = append(opts, grpc.WithBlock())
```

Comment:
This is discouraged by the grpc library https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md

However, it can make sense to use in test scenarios--as long as we don't include this in production configurations I'm fine with it.